# 06 - Business Insights, Experimentation & Recommendations

In [1]:
import sys; sys.path.insert(0, '../src')
from data_preprocessing import run as preprocess
import experiment_design as ed

df, _ = preprocess('../data/coffee_shop_sales.csv')
result = ed.run(df)
result

{'power_analysis': {'baseline_mean_revenue': np.float64(4.6864),
  'baseline_std_revenue': np.float64(4.2271),
  'assumed_uplift_pct': 0.05,
  'cohens_d': np.float64(0.0554),
  'alpha': 0.05,
  'power': 0.8,
  'required_n_per_arm': 5110},
 'simulated_ab_test': {'note': 'SIMULATED experiment - uplift was injected, not observed in real Maven Roasters/Yum data.',
  'n_per_arm': 4000,
  'control_mean': np.float64(4.759),
  'treatment_mean': np.float64(5.1085),
  'observed_lift_pct': np.float64(7.34),
  't_stat': np.float64(2.3125),
  'p_value': np.float64(0.02078748002226551),
  '95pct_ci_diff': (np.float64(0.0533), np.float64(0.6456)),
  'significant_at_0.05': True}}

**Reading the experiment design output**

- `power_analysis` says a promo experiment aiming to detect a *5% lift in average transaction value* needs about **5,110 transactions per arm** at 80% power / 5% significance - useful for telling a stakeholder how long a real test would need to run before trusting the result.
- `simulated_ab_test` is a synthetic experiment (uplift is injected, not observed) that exercises the same t-test / confidence-interval workflow a real promo readout would use. It is run at 4,000/arm - below the required 5,110 - which is a deliberate illustration of why under-powering a test is risky even when it happens to come back significant.

**Business recommendations (from the real analysis above)**
1. Treat Coffee and Tea as the demand-forecasting priority - they drive ~66% of revenue, so a 1-point MAPE improvement there is worth more than the same improvement on Branded or Packaged Chocolate.
2. Use recent rolling demand (7-14 day average), not the calendar, as the main forecasting signal - that is what the model actually leans on.
3. Before running a real promo, size the test with the power analysis above rather than picking an arbitrary sample/duration.
4. Investigate why Lower Manhattan's average basket is higher (menu mix vs. pricing vs. customer type) before copying its promo strategy to Astoria - the ANOVA says the stores are not interchangeable.